include/flatbuffers/code_generators.h

```diff
@@ -74,9 +74,35 @@ class BaseGenerator {
     return true;
   }
 
-  std::string FullNamespace(const char *separator, const Namespace &ns) {
+  static void NamespacePrefixAsComponents(const std::string &prefix, std::vector<std::string> &namespaces)
+  {
+    std::string::size_type componentStart = 0;
+    std::string::size_type componentEnd = prefix.find_first_of('.', componentStart);
+    while (componentEnd != std::string::npos) {
+      namespaces.push_back(prefix.substr(componentStart, componentEnd - componentStart));
+      componentStart = componentEnd + 1;
+      componentEnd = prefix.find_first_of('.', componentStart);
+    }
+    if (componentStart < prefix.length())
+      namespaces.push_back(prefix.substr(componentStart, prefix.length() - componentStart));
+  }
+
+  // Returns namespace components, including the language specific prefix splitted by '.'
+  static std::vector<std::string> PrefixedNamespaceComponents(const Namespace &ns, IDLOptions::Language language)
```

Comment:
Why not pass the vector as argument like the previous function? that avoids potentially costly copying.